online-shop/src/app/product-detail/product-detail.component.html

```diff
@@ -3,6 +3,7 @@
   (addToCartEvent)="onAddToCart()"
   (deleteEvent)="onDelete()"
   (editEvent)="activateEdit()"
+  [authService]="authService"
```

Comment:
same idea as before, instead of sending in the whole auth service, add some flags on the presentational component that dictate if it should show/hide certain parts of the app. either one isAdmin flag (sufficient here) or 3 flags canDelete, canEdit, canAddToCart (way more granular, not needed in this case) and you pass in the already computed values